src/rate_limiter/src/lib.rs

```diff
@@ -347,6 +358,23 @@ impl RateLimiter {
         })
     }
 
+    // Arm the timer of the rate limiter with `duration` milliseconds.
+    // If `duration` is `None`, the minimum refill duration will be used.
+    fn activate_timer(&mut self, duration: Option<u64>) {
+        if !self.timer_active {
```

Comment:
I think we can drop this check. Timer should never be active when we get here, and even if it was, we should update it to its new deadline.
```suggestion
```

Reply:
It should indeed overwrite the older timer config anyway.